presto-main/src/main/java/io/prestosql/sql/planner/PlanFragmenter.java

```diff
@@ -604,6 +608,33 @@ public GroupedExecutionProperties visitAggregation(AggregationNode node, Void co
             return GroupedExecutionProperties.notCapable();
         }
 
+        @Override
+        public GroupedExecutionProperties visitWindow(WindowNode node, Void context)
+        {
+            return visitWindowFunctionNode(node);
+        }
+
+        @Override
+        public GroupedExecutionProperties visitRowNumber(RowNumberNode node, Void context)
+        {
+            return visitWindowFunctionNode(node);
+        }
+
+        @Override
+        public GroupedExecutionProperties visitTopNRowNumber(TopNRowNumberNode node, Void context)
+        {
+            return visitWindowFunctionNode(node);
+        }
+
+        private GroupedExecutionProperties visitWindowFunctionNode(PlanNode node)
+        {
+            GroupedExecutionProperties properties = getOnlyElement(node.getSources()).accept(this, null);
+            if (groupedExecutionForAggregation && properties.isCurrentNodeCapable()) {
```

Comment:
We shouldn't overload the "groupedExecutionForAggregation" flag to control the behavior for window functions. We should either create a new option or generalize the existing one.